COMPACT_DENYLIST_FORMAT.md

```diff
@@ -0,0 +1,457 @@
+
+# Block all mime types with exceptions
+/mime/image/*
+!/mime/image/gif
+
+# Legacy CID double-hash block
+# sha256(bafybeiefwqslmf6zyyrxodaxx4vwqircuxpza5ri45ws3y5a62ypxti42e/)
+# blocks only this CID
+//d9d295bde21f422d471a90f2a37ec53049fdf3e5fa3ee2e8f20e10003da429e7
+
+# Legacy Path double-hash block
+# Blocks bafybeiefwqslmf6zyyrxodaxx4vwqircuxpza5ri45ws3y5a62ypxti42e/path
+# but not any other paths.
+//3f8b9febd851873b3774b937cce126910699ceac56e72e64b866f8e258d09572
+
+# Double hash CID block
+# base58btc-sha256-multihash(QmVTF1yEejXd9iMgoRTFDxBv7HAz9kuZcQNBzHrceuK9HR)
+# Blocks bafybeidjwik6im54nrpfg7osdvmx7zojl5oaxqel5cmsz46iuelwf5acja
+# and QmVTF1yEejXd9iMgoRTFDxBv7HAz9kuZcQNBzHrceuK9HR etc. by multihash
+//QmX9dhRcQcKUw3Ws8485T5a9dtjrSCQaUAHnG4iK9i4ceM
+
+# Double hash Path block using blake3 hashing
+# base58btc-blake3-multihash(gW7Nhu4HrfDtphEivm3Z9NNE7gpdh5Tga8g6JNZc1S8E47/path)
+# Blocks /ipfs/bafyb4ieqht3b2rssdmc7sjv2cy2gfdilxkfh7623nvndziyqnawkmo266a/path
+# /ipfs/bafyb4ieqht3b2rssdmc7sjv2cy2gfdilxkfh7623nvndziyqnawkmo266a/path
+# /ipfs/f01701e20903cf61d46521b05f926ba1634628d0bba8a7ffb5b6d5a3ca310682ca63b5ef0/path etc...
+# But not /path2
+//QmbK7LDv5NNBvYQzNfm2eED17SNLt1yNMapcUhSuNLgkqz
+```
+
+#### High level list format
+
+A denylist is made of an optional header and a list of blockitems separated by newlines. Comment lines start with `#`. Empty lines are allowed.
+
+```
+<header>
+---
+<block_item1> [hint_list]
+
+# comment
+<block_item2> [hint_list]
+...
+```
+
+#### Header
+
+The list header is a YAML block:
+
+- Must be valid YAML
+- Fully optional
+- 1KB maximum size
+- Delimited by a line containing `---` at the end (document separator)
+
+Known-fields (they must be lowercase):
+
+- `version`: the denylist format version. Defaults to 1 when not specified.
+- `name`
+- `description`
+- `author`
+- `hints`: a map of *hints*. See section below for known hints
+
+#### Hints
+
+A *hint* is a key-value duple associated to the denylist as a whole (part of the header), or to a specific \<block_item\>.
+
+Header hints can be used to set denylist-wide options or information that
+implementations can choose to interpret or not.
+
+#### List body
+
+A denylist is made of lines which are made by a *block items* followed by zero or more space-separated hints.
+
+Lines should not be longer than 2MiB including the "\n" delimiter.
+
+#### Block item
+
+A block item represents a rule to enable content-blocking:
+
+- `PATH` elements are expected to be %-encoded, per [RFC 3986, section 2.1](https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding).
+- `CID` elements represent a CID (either V0 or V1).
+- `CIDv0` are for us equivalent to baseb58btc-encoded sha256 multihashes although they are not the same thing (a CIDV0 carries implicit codec (dag-pb) and multibase information (b58btc). When we say a b58-encoded multihash needs to be extracted from the CID, this usually is a no-op in case of CIDv0s.
```

Comment:
Is there specific processing that implementations are expected to carry out for items they don't recognise? Skipping is more resilient to format evolution in a distributed system but I would worry that operators might think they're properly protected by tracking a given list with legal implications when in fact their deployment is skipping over some items it doesn't get.

Reply:
Implementation-specific? I'm not sure the specification should say how implementations react to things outside the spec. You could kill everything, or you could log an error message, or you could log a warn message. Perhaps we can recommend logging an error.

Reply:
Added a note